### PR TITLE
add redis appender

### PR DESCRIPTION
**Pull request by wadecha**
redis appender support redis publish message. log receiver server receive the message when it subscribe channel.

### CODE REVIEW

**Comment by wadecha** [comment]
I don't know why check failed, the error note is 'error  Parsing error: Unexpected token'.  I use ES5.1, so I don't know how correct the code.

File: lib/appenders/redis.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var layouts = require('../layouts');
```
Comment (nomiddlename) [inline lib/appenders/redis.js]: change `var` to `const` to get rid of the eslint errors you're seeing.

File: lib/appenders/redis.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    redisClient.publish(channel, message, function (err) {
+      if (err) {
+        console.error('log4js.redisAppender - %s:%p Error: %s', host, port, util.inspect(err));
+      }
+    });
+  };
+}
+
+function configure(config) {
+  var layout, host, port, pass, channel;
```
Comment (nomiddlename) [inline lib/appenders/redis.js]: Replace this `var` with `let`.

**Comment by wadecha** [comment]
The eslint check passed, but I don't know the other two checks error, can you help me?

**Comment by nomiddlename** [comment]
The other two checks are failing because you haven't added any unit tests that cover this appender. I can't accept the MR until there are some tests, sorry.

**Comment by wadecha** [comment]
HI, I am add the unit test

File: package.json [hunk 3]
```diff
@@ -39,6 +39,7 @@
   "dependencies": {
     "date-format": "^1.0.0",
     "debug": "^2.2.0",
+    "redis": "^2.6.5",
```
Comment (nomiddlename) [inline package.json]: don't put redis in dependencies - put it in "optionalDependencies" - not every user will need to use the redis appender.

**Comment by nomiddlename** [comment]
Thanks for adding the tests - looks good. Just that one change to package.json and we should be good. Bu the way, did you know that you can run all the tests and validations locally, without having to push to github every time? It looks like you were making changes and pushing to github to see if they worked. `npm test` will run the lint and tests, `npm run coverage` will output the coverage report.

**Comment by wadecha** [comment]
I am sorry, I know it later, so download the project and run it in local.

**Comment by nomiddlename** [comment]
Thanks for this. I've merged it, but it won't get out to npm until I've got version 2 done.

**Pull request merged**